File: readability.py
text = input("Text: ")
letters = 0
words = 1
sentences = 0
for i in range(0, len(text), 1):
    if (text[i] >= 'a' and text[i] <= 'z') or (text[i] >= 'A' and text[i] <= 'Z'):
        letters += 1
    elif (text[i] == ' '):
        words += 1
    elif (text[i] == '.' or text[i] == '!' or text[i] == '?'):
        sentences += 1
level = round(0.0588 * letters / words * 100 - 0.296 * sentences / words * 100 - 15.8)
if (level < 0):
    print("Before Grade 1")
elif (level >= 16):
    print("Grade 16+")
else:
    print(f"Grade {level}")

File: Week2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string text = get_string("Text: ");
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);
    int index = round(0.0588 * letters / words * 100 - 0.296 * sentences / words * 100 - 15.8);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}
int count_letters(string text)
{
    int count = 0;
    int i = 0;
    while (text[i] != '\0')
    {
        if ((text[i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z'))
        {
            count++;
        }
        i++;
    }
    return count;
}
int count_words(string text)
{
    int count = 1;
    int i = 0;
    while (text[i] != '\0')
    {
        if (text[i] == ' ')
        {
            count++;
        }
        i++;
    }
    return count;
}
int count_sentences(string text)
{
    int count = 0;
    int i = 0;
    while (text[i] != '\0')
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            count++;
        }
        i++;
    }
    return count;
}